applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp

```diff
@@ -0,0 +1,326 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Inigo Lopez
+//
+
+// Project includes
+#include "define_2d_wake_process.h"
+#include "utilities/variable_utils.h"
+#include "utilities/openmp_utils.h"
+
+namespace Kratos {
+
+// Constructor for Define2DWakeProcess Process
+Define2DWakeProcess::Define2DWakeProcess(ModelPart& rBodyModelPart, const double Tolerance)
+    : Process(), mrBodyModelPart(rBodyModelPart), mTolerance(Tolerance)
+{
+    ModelPart& root_model_part = mrBodyModelPart.GetRootModelPart();
+    root_model_part.CreateSubModelPart("trailing_edge_sub_model_part");
+}
+
+void Define2DWakeProcess::ExecuteInitialize()
+{
+    // Initialize element and node variables for the process
+    //InitializeVariables();
```

Comment:
This line is commented!

Reply:
In short: I know, I need to find a solution for this for FSI.

I commented it on purpose because I saw a large detriment in the performance when initializing the variables for all elements and nodes. In general initializing everywhere is not needed and brings overhead. In FSI we need to either initialize or clear the variables corresponding to the wake, kutta and trailing edge elements from one iteration to another. What about for MonteCarlo? Do you have any requirements with respect to the initialization of the variables? (Right now the python version in master doesn't do it and at least the primal and adjoint analysis work fine, I don't know about MLMC analysis)

Reply:
> 
> In FSI we need to either initialize or clear the variables corresponding to the wake, kutta and trailing edge elements from one iteration to another.

We need the same in MLMC, since the angle of attack is changing for every sample so the wake is slightly different every time. For now what I was doing was adding the clearing/resetting in the python process (I only did this in custom branches). 

I would propose to add a boolean parameter to turn on/off the initizalization of variables.

Another option is to reset the variables in the same loop that we mark the wake elements:

https://github.com/KratosMultiphysics/Kratos/blob/beeea49b84aaccf71cab9171fdc8f4e1d98646e1/applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp#L114-L145

Here at line 116 we could reset all element variables in the same loop. Not sure if that would be faster than using VariableUtils()

Reply:
I see, a question for my understanding, I thought that for MLMC you launch independent analysis with different models however from this discussion I understand that all the processes you launch in MLMC share the same model (which is why you require to clean the submodelparts and variables for each run like in FSI), is that right?

In any case, before reading your comment I came up with the following solution:

https://github.com/KratosMultiphysics/Kratos/blob/af4b2c0903699afdb3535201a003f9998904e0b1/applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp#L41-L90

What do you think? As you can see, here I do the check that you suggested above and the resetting is done only for all element variables from the submodelparts, which does not introduce as much overhead as resetting the variables of all the elements in the root modelpart.

Reply:
Hi! For the case of MonteCarlo, yes they are independant! However in MultiLevelMontecarlo, more than one mesh is used, and each mesh is generated from the previous one by remeshing according to the previous solution (these are called the levels). 

So when solving several levels, the same model is used for a given sample (since this model is the one that is going through solver -> remeshing -> solver -> remeshing...and so on), and thats when we get the flag/modelpart problem.

Your implementation seems fine to me, thanks!

Reply:
I see! I was mixing MonteCarlo with MultiLevelMontecarlo. Thanks for the explanation!